src/oracle/middleware.js

```diff
@@ -0,0 +1,761 @@
+    var currentOffer = offers[i];
+
+    //if a price invitation and for the correct feed
+    if (currentOffer["status"]["invitationSpec"]["invitationMakerName"] == "PushPrice" && currentOffer["status"]["invitationSpec"]["previousOffer"] == feedOfferId) {
+      let offerRound = Number(currentOffer["status"]["invitationSpec"]["invitationArgs"][0]["roundId"])
+
+      //if it is an offer for the round we are checking for
+      if (offerRound == roundId) {
+        //if there is no error
+        if (!currentOffer["status"].hasOwnProperty("error")) {
+          return true
+        }
+      }
+      //else if offer round id is less than the round we want to check and its satisfied
+      //return false because there cannot be a submission for a newer round before this offer 
+      else if (offerRound < roundId && !currentOffer["status"].hasOwnProperty("error")) {
+        return false
+      }
+    }
+  }
+  return false
+}
+
+/**
+  * Function to query price from chain
+  * @param {*} feed feed name of the price to query in the form of ATOM-USD
+  * @returns the latest price
+  */
+export const queryPrice = async (feed) => {
+
+  try {
+    //read value from vstorage
+    const capDataStr = await readVStorage(vstorage, feed, false);
+
+    //parse the value
+    var capData = JSON.parse(JSON.parse(capDataStr).value)
+    capData = JSON.parse(capData.values[0])
+    //replace any extra characters
+    capData = JSON.parse(capData.body.replaceAll("\\", ""))
+
+    //get the latest price by dividing amountOut by amountIn
+    var latestPrice = Number(capData.amountOut.value.digits) / Number(capData.amountIn.value.digits)
+
+    console.log(feed + " Price Query: " + String(latestPrice))
+    return latestPrice
+  }
+  catch {
+    return 0
+  }
+}
+
+/**
+ * Function to get oracles feed invitations
+ * @returns an object containing feed invitation IDs
+ */
+export const getOraclesInvitations = async () => {
+
+  //if agoric names do not exist, create them
+  if (Object.keys(agoricNames).length == 0){
+    agoricNames = await makeAgoricNames(fromBoard, vstorage);
+  }
+
+  let feedBoards = agoricNames.reverse
+
+  let feedInvs = {}
+
+  const current = await getCurrent(String(FROM), fromBoard, { vstorage });
+  const invitations = current.offerToUsedInvitation
+
+  //for each invitation
+  for (let inv in invitations) {
+    let boardId = invitations[inv].value[0].instance.boardId
+    let feed = feedBoards[boardId].split(" price feed")[0]
+
+    feedInvs[feed] = Number(inv)
+  }
+
+  return feedInvs
+}
+
+/**
+  * Function to query round from chain
+  * @param {*} feed feed name of the price to query in the form of ATOM-USD
+  * @returns the latest round
+  */
+export const queryRound = async (feed) => {
+
+  //read value from vstorage
+  const capDataStr = await readVStorage(vstorage, feed, true);
+
+  //parse the value
+  var capData = JSON.parse(JSON.parse(capDataStr).value)
+  capData = JSON.parse(capData.values[capData.values.length - 1])
```

Comment:
Why removing the last value, something that is not clear should be explained in comments.

The comment "//parse the value", I can get it immediately from looking at the code so the comment is a little redundant unless you want to use it to break code into segments but [capData.values.length - 1] i dont know why you're removing the last value